WordPress/src/main/java/org/wordpress/android/util/URLFilteredWebViewClient.java

```diff
@@ -35,10 +45,37 @@ public URLFilteredWebViewClient(Collection<String> urls) {
         mAllowedURLs.addAll(urls);
     }
 
+    public void setListener(URLWebViewClientListener listener) {
+        mListener = listener;
+    }
+
     protected boolean isAllURLsAllowed() {
         return mAllowedURLs.size() == 0;
     }
 
+    @Override
+    public void onPageStarted(WebView view, String url, Bitmap favicon) {
+        super.onPageStarted(view, url, favicon);
+        mWebResourceError = false;
+    }
+
+    @Override
+    public void onPageFinished(WebView view, String url) {
+        super.onPageFinished(view, url);
+        if (mListener != null && !mWebResourceError) {
+            mListener.onPageLoaded();
+        }
+    }
+
+    @Override
+    public void onReceivedError(WebView view, WebResourceRequest request, WebResourceError error) {
```

Comment:
I read the [onReceivedError documentation](https://developer.android.com/reference/android/webkit/WebViewClient.html#onReceivedError(android.webkit.WebView,%2520android.webkit.WebResourceRequest,%2520android.webkit.WebResourceError)) and found:
> These errors usually indicate inability to connect to the server. Note that unlike the deprecated version of the callback, the new version will be called for any resource (iframe, image, etc.), not just for the main page. Thus, it is recommended to perform minimum required work in this callback.

This might be a bad idea to trigger the full error screen if only a part of the page is not loading (e.g. if there is a missing image, I think it's better to show the page than the error)

Reply:
That is a great point Maxime. I just did some more digging and found out that WebResourceRequest has a `isForMainFrame()` method which tells whether the request was made for the main frame, so I'll be adding that as a check.